azure-kusto-data/azure/kusto/data/client.py

```diff
@@ -853,19 +853,61 @@ def execute_streaming_ingest(
 
         self._execute(endpoint, database, None, stream, self._streaming_ingest_default_timeout, properties)
 
-    def _execute(
-        self, endpoint: str, database: str, query: Optional[str], payload: Optional[io.IOBase], timeout: timedelta, properties: ClientRequestProperties = None
+    def execute_streaming_query(
+        self, database: str, query: str, timeout: timedelta = _KustoClientBase._query_default_timeout, properties: Optional[ClientRequestProperties] = None
+    ) -> Response:
+        """
+        Query directly from Kusto database using streaming output.</p>
+        This method queries the Kusto database into an asyncio stream.
+
+        example usage::
+
+            my_file = open("some_file", "wb")
+            with client.execute_streaming_query(...) as response:
+                for chunk in response.iter_content(1024):
+                    file.write(chunk)
+
+        :return: a Response that can be read as chunks
+        """
+        response = self._retrieve_response(self._query_endpoint, database, properties, query, None, timeout, stream=True)
+
+        try:
+            response.raise_for_status()
+        except Exception as e:
+            self._handle_http_error(e, self._query_endpoint, None, response, response.json(), response.text)
+
+        return response
+
+    def _retrieve_response(

```

Comment:
Can't it be part of _execute?

Reply:
It's for sharing code between _execute and execute_streaming_query

Reply:
_execute was the method for sharing code between all the "public" methods - so now its not good enough so we add _execute2? - I don't like it